main.go

```diff
@@ -52,7 +53,45 @@ func main() {
 		log.Fatal(errors.AddContext(err, "failed to build the api"))
 	}
 
-	logger.Printf("Starting Pinner service.\nGitRevision: %v (built %v)\n", build.GitRevision, build.BuildTime)
+	logger.Print("Starting Pinner service")
+	logger.Printf("GitRevision: %v (built %v)", build.GitRevision, build.BuildTime)
 	err = server.ListenAndServe(4000)
 	log.Fatal(errors.Compose(err, scanner.Close()))
 }
+
+// newLogger creates a new logger that can write to disk.
+//
+// The function also returns a closer function that should be called when we
+// stop using the logger, typically deferred in main.
+func newLogger(level, logfile string) (logger *logrus.Logger, closer func(), err error) {
```

Comment:
I thought about that. But having in mind that we'd never have anything else to do with it other than just log it, it felt like unnecessary complexity that I could neatly fold into a lower level.

Reply:
Ok, even if when I don't really need it I usually just write `closeFn func() error` because you expect an error from a close function 🤷‍♂️ . But it doesn't matter much I guess it's just PP